js/parts-3d/Column.js

```diff
@@ -339,16 +349,35 @@ wrap(Series.prototype, 'alignDataLabel', function (proceed) {
 // Added stackLabels position calculation for 3D charts.
 wrap(H.StackItem.prototype, 'getStackBox', function (proceed, chart) {
     var stackBox = proceed.apply(this, [].slice.call(arguments, 1));
-    // Only do this for 3D chart.
+    // Only do this for 3D column and iherited series.
     if (chart.is3d()) {
-        var pos = ({
-            x: stackBox.x,
-            y: stackBox.y,
-            z: 0
+        var columnSeries;
+        // Check if any series iherits from column 3D
+        chart.series.forEach(function (series) {
+            if (series instanceof seriesTypes.column && series.data.length) {
+                columnSeries = series;
+            }
         });
-        pos = H.perspective([pos], chart, true)[0];
-        stackBox.x = pos.x;
-        stackBox.y = pos.y;
+        // If any series is a column series, use its barW, z and depth
+        // parameters for correct stackLabels position calculation
+        if (columnSeries) {
```

Comment:
Why this If is not in forEach from line 356?

Reply:
The reason was that we wanted to find any column series in the graph and then assign it to columnSeries variable. However, this part of code is already refactored so we can mark it as resolved.